src/components/UserManagement.tsx

```diff
@@ -6,23 +6,192 @@
  * purpose: page that wiill provide access to manage users
  */
 
-import { getUserPrivilege } from '../scripts/Datastore';
+import {
+    deleteUser,
+    editEmail,
+    editPhoneNumber,
+    editPrivilege,
+    getUserPrivilege,
+    registerUser,
+} from '../scripts/Datastore';
 import AuthCheck from './AuthCheck';
-import React, { useState } from 'react';
 import Content from './Content';
+import { useState } from 'react';
+import ReactDataGrid from '@inovua/reactdatagrid-community';
+import '@inovua/reactdatagrid-community/index.css';
+import { useSelector } from 'react-redux';
+import { RootState } from 'store/rootReducer';
+import { User } from 'store/FirestoreInterfaces';
+import { TypeEditInfo } from '@inovua/reactdatagrid-community/types';
+import Button, { ButtonType } from './Button';
+
+interface PrivilegeToggleProps {
+    level: string;
+    uid: string;
+}
+
+function PrivilegeToggle(props: PrivilegeToggleProps) {
+    return (
+        <div className="privilegeToggle">
+            <span
+                onClick={() => {
```

Comment:
These spans are the same except for the user privilege string, so this can be made into a function.